scripts/setDefaultIndex.py

```diff
@@ -45,11 +47,22 @@ def add_missing_elements_to_dict(to_verify, original_content):
             missing_elements[key] = original_content[key]
     return missing_elements
 
+def flatten_dict(d):
+    def items():
+        for key, value in d.items():
+            if isinstance(value, dict):
+                for subkey, subvalue in flatten_dict(value).items():
+                    yield key + "." + subkey, subvalue
```

Comment:
please add code comment regarding when you need dotted notation 
vs
line 57 when you do not return dotted notation.